### PR TITLE
gRPC treating clients without configuration as self-calling

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
At the moment, gRPC testing experience is worse than REST. A user has to know the port of the gRPC server in tests, and has to configure a client stub for tests in application.properties.

This PR proposes allowing injection of client stubs without config when running in test mode. Such clients are automatically configured to call the exposed gRPC server.

The drawback of this change is that misconfiguration problems may be a bit more difficult to spot in tests.

@cescoffier @mkouba WDYT?

### CODE REVIEW

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
      */
     String value() default ELEMENT_NAME;
 
-    public final class Literal extends AnnotationLiteral<GrpcClient> implements GrpcClient {
+    final class Literal extends AnnotationLiteral<GrpcClient> implements GrpcClient {
```
Comment (mkouba) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java:33]: Why this change?

**Review by mkouba — COMMENTED** [review]
This feature is intended for internal tests (aka `QuarkusUnitTest`) anyway, or?

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java:33]: everything is public in an interface anyway

**Comment by michalszynkiewicz** [comment]
It should work with `@QuarkusTest` too, no?

Native may not work though, I'm not sure if we enable test profile when running a native executable.

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java (hunk 1 above)
Comment (mkouba) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/GrpcClient.java:33]: Ah, I didn't know it works like this for annotations as well.

**Review by cescoffier — APPROVED** [review]
